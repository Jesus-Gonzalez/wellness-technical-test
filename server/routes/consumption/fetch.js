import Models from '~/database/models'

export const consumptionFetchRoute = app => {
  app.get('/consumption', (req, res, next) => {
    Models.ConsumptionModel
      .find().then(consumptions => {
        res
          .status(200)
          .type('application/json')
          .end(JSON.stringify(consumptions))
      })
      .catch(error => {
        res.status(500).end(error)
      })
      .finally(() => {
        next()
      })
  })
}
