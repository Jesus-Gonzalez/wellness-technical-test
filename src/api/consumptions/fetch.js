import { HttpClient } from 'api/http'

export const fetchConsumptions = data => (
  HttpClient({
    url: '/consumption',
    method: 'get',
    data
  }).then(({ data }) => data.sort((a, b) => a.date > b.date ? 1 : -1))
)
