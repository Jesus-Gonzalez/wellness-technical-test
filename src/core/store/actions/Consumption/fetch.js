import * as ActionTypes from 'core/store/actionTypes'

export const fetchConsumptions = () => ({
  type: ActionTypes.Consumption.Fetch
})
