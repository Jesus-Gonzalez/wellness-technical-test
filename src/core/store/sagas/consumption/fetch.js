import { call, put, takeLatest } from 'redux-saga/effects'

import * as ActionTypes from 'core/store/actionTypes'
import * as Api from 'api'

function* fetchConsumptions() {
  console.log('fetching consumptions')

  try {
    const items = yield call(Api.fetchConsumptions)
    yield put({ type: ActionTypes.Consumption.FetchSuccess, items })
  } catch {
    yield put({ type: ActionTypes.Consumption.FetchError })
  }
}

export default function* fetchConsumptionsSaga() {
  yield takeLatest(ActionTypes.Consumption.Fetch, fetchConsumptions)
}
