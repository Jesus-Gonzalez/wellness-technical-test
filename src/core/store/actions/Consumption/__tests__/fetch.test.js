import { fetchConsumptions } from '../fetch'

describe('redux', () => {
  describe('actions', () => {
    describe('fetch', () => {
      it('should return an action', () => {
        expect(fetchConsumptions()).toMatchSnapshot()
      })
    })
  })
})
