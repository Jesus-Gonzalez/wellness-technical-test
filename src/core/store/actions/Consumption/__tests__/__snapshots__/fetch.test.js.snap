// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux actions fetch should return an action 1`] = `
Object {
  "type": "FETCH_CONSUMPTIONS",
}
`;
